proxy.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/cloudflare/unsee/internal/alertmanager"
+	"github.com/gin-gonic/gin"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func newProxyServer(domain string) *httputil.ReverseProxy {
+	url, err := url.Parse(domain)
+	if err != nil {
+		log.Error(err)
+	}
+	proxy := httputil.NewSingleHostReverseProxy(url)
+	proxy.ModifyResponse = func(resp *http.Response) error {
+		// drop Content-Length header from upstream responses, gzip middleware
+		// will compress those and that could cause a mismatch
+		resp.Header.Del("Content-Length")
+		return nil
+	}
+	return proxy
+}
+
+func handleAlertmanagerProxyRequest(name string, c *gin.Context) {
+	upstream := alertmanager.GetAlertmanagerByName(name)
+	if upstream == nil {
+		c.String(http.StatusNotFound, "No Alertmanager instance named '%s'\n", name)
+	} else if !upstream.ProxyRequests {
+		c.String(http.StatusBadRequest, "Alertmanager instance named '%s' does not allow proxying\n", name)
+	} else {
+		proxy := newProxyServer(upstream.URI)
```

Comment:
You probably don't want to create a whole new proxy for each request. An alternative architecture could rely on the fact that `ReverseProxy` satisfies the `http.Handler` interface, and that you can add an `http.Handler` to a Gin router with `gin.WrapH(http.Handler) gin.HandlerFunc`.  Then can direct the two routes to the upstream by implementing `ReverseProxy.Director`.

Reply:
Updated, is this better? I also got some tests this time